xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/hibernate/query/HqlQueryExecutor.java

```diff
@@ -143,17 +159,46 @@ protected static boolean isSafeSelect(String statementString)
         return HqlQueryUtils.isSafe(completeStatement);
     }
 
+    private boolean isSafe(String statement) throws QueryException
+    {
+        // An empty statement is safe
+        if (statement.isEmpty()) {
+            return true;
+        }
+
+        // HqlQueryUtils#isSafe only works with complete statements
+        String completeStatement = toCompleteShortForm(statement);
+
+        // Parse and validate the statement
+        ComponentManager componentManager = this.componentManagerProvider.get();
+        try {
+            for (HQLQueryValidator validator : componentManager
+                .<HQLQueryValidator>getInstanceList(HQLQueryValidator.class)) {
+                Optional<Boolean> result = validator.isSafe(completeStatement);
+                if (result.isPresent()) {
```

Comment:
Doesn't this mean that if for whatever reason the `DefaultHQLQueryValidator` is first in the component list, the configuration will be ignored? Shouldn't we use priorities to define an explicit order between them? Or are they ordered alphabetically by default? In any case, I think it would be important to document this ordering requirement, even for an internal API.

Reply:
> Shouldn't we use priorities to define an explicit order between them?

Yes, it was the plan. It's for this kind of missing details that I still a draft that I need to review.